python/lsst/pipe/tasks/selectImages.py

```diff
@@ -590,3 +593,246 @@ def run(self, wcsList, bboxList, coordList, psfList, dataIds, **kwargs):
+                       zip(detectorSummary['raCorners'], detectorSummary['decCorners'])]
+            detectorPolygon = lsst.sphgeom.ConvexPolygon.convexHull(corners)
+            if detectorPolygon.intersects(polygon):
+                doesIntersect = True
+                break
+        return doesIntersect
+
+
+class BestSeeingQuantileSelectVisitsConfig(pipeBase.PipelineTaskConfig,
+                                           pipelineConnections=BestSeeingSelectVisitsConnections):
+    qMin = pexConfig.RangeField(
+        doc="Lower bound of quantile range to select. Sorts visits by seeing from narrow to wide, "
+            "and select those in the interquantile range (qMin, qMax). Set qMin to 0 for Best Seeing. "
+            "This config should be changed from zero only for exploratory diffIm testing.",
+        dtype=float,
+        default=0,
+        min=0,
+        max=1,
+    )
+    qMax = pexConfig.RangeField(
+        doc="Upper bound of quantile range to select. Sorts visits by seeing from narrow to wide, "
+            "and select those in the interquantile range (qMin, qMax). Set qMax to 1 for Worst Seeing.",
+        dtype=float,
+        default=0.33,
+        min=0,
+        max=1,
+    )
+    nVisitsMin = pexConfig.Field(
+        doc="At least this number of visits selected and supercedes quantile. For example, if 10 visits "
+            "cover this patch, qMin=0.33, and nVisitsMin=5, the best 5 visits will be selected.",
+        dtype=int,
+        default=3,
+    )
+    doConfirmOverlap = pexConfig.Field(
+        dtype=bool,
+        doc="Do remove visits that do not actually overlap the patch?",
+        default=True,
+    )
+
+
+class BestSeeingQuantileSelectVisitsTask(BestSeeingSelectVisitsTask):
+    """Select a quantile of the best-seeing visits
+
+    Selects the best (for example, third) full visits based on the average
+    PSF width in the entire visit. It can also be used for difference imaging
+    experiments that require templates with the worst seeing visits.
+    For example, selecting the worst third can be acheived by
+    changing the config parameters qMin to 0.66 and qMax to 1.
+    """
+    ConfigClass = BestSeeingQuantileSelectVisitsConfig
+    _DefaultName = 'bestSeeingQuantileSelectVisits'
+
+    @utils.inheritDoc(BestSeeingSelectVisitsTask)
+    def run(self, visitSummaries, skyMap, dataId):
+        if self.config.doConfirmOverlap:
+            patchPolygon = self.makePatchPolygon(skyMap, dataId)
+        visits = np.array([visitSummary.ref.dataId['visit'] for visitSummary in visitSummaries])
+        radius = np.empty(len(visits))
+        intersects = np.full(len(visits), True)
+        for i, visitSummary in enumerate(visitSummaries):
+            # read in one by one; and only once. There may be hundreds
+            visitSummary = visitSummary.get()
+            psfSigma = np.nanmedian([vs['psfSigma'] for vs in visitSummary])
```

Comment:
Please add the comment you have up above here too for clarity
`# psfSigma is PSF model determinant radius at chip center in pixels`